### PR TITLE
Background Audio: Support non-looping audio

### LINKED ISSUE #10010: Background Audio: Support non-looping audio
## Feature Description

Some creators are using background audio to narrate the story.

They would like the page level background audio to be non-looping and to be considered for the "auto" page advancement calculation.

For this to work we'd need to:

* Store the audio length together with the URL when saving the audio
* Add background audio as an `<amp-video` element instead of with a `background-audio` attribute
* Update the advancement calculation to take into account the page background audio

## Additional Context

**Note:** narration also means there's a need for captions, hence the idea of using `<amp-video>` for audio. See #8821

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add an option to make background audio on pages, none looping. 

With this change, force the audio advance to match length of the longer media or the background audio. 

## Relevant Technical Choices

Reuse the same loop checkbox found on videos. 
Only show looping option if audio length is available. 
Moved logic for generating auto advance in it's own function, to improve testability and keep complexity down. 

## To-do

UX question - https://github.com/GoogleForCreators/web-stories-wp/issues/10010#issuecomment-1021452975

## User-facing changes

Screenshot 2022-01-28 at 15 06 45

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Enable background audio experiment
2. Create story.
3. Add background audio to a page. 
4. Uncheck looping. 

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10010

### CODE REVIEW

File: packages/story-editor/src/output/utils/test/getAutoAdvanceAfter.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    );
+  });
+
+  it('should return default value', () => {
+    expect(getAutoAdvanceAfter({ elements: [], defaultPageDuration: 7 })).toBe(
+      '7s'
+    );
+  });
+
+  it('should return background audio length', () => {
```
Comment (swissspidy) [inline packages/story-editor/src/output/utils/test/getAutoAdvanceAfter.js:42]: This should return the element ID of the background audio element, not its length

Comment (spacedmonkey) [reply packages/story-editor/src/output/utils/test/getAutoAdvanceAfter.js:42]: Fixed in https://github.com/GoogleForCreators/web-stories-wp/pull/10401/commits/c85ad0d8265162707c8e45f028ca465eacca61dc

File: packages/story-editor/src/output/page.js [hunk 2]
```diff
@@ -98,15 +93,20 @@ function OutputPage({ page, autoAdvance = true, defaultPageDuration = 7 }) {
     videoCaptions.push(`el-${id}-captions`);
   }
 
+  const isNotLoopingBackgroundAudio =
```
Comment (swissspidy) [inline packages/story-editor/src/output/page.js]: Nit: `isNonLoopingBackgroundAudio`

File: packages/story-editor/src/output/utils/getAutoAdvanceAfter.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * External dependencies
+ */
+import { getTotalDuration } from '@googleforcreators/animation';
+/**
+ * Internal dependencies
+ */
+import getLongestMediaElement from './getLongestMediaElement';
+
+function getAutoAdvanceAfter({
```
Comment (swissspidy) [inline packages/story-editor/src/output/utils/getAutoAdvanceAfter.js:26]: Good idea to extract this and make it more testable 👍 

Note that there's a lint error in this file

File: packages/story-editor/src/output/utils/test/getAutoAdvanceAfter.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies
+ */
+import getAutoAdvanceAfter from '../getAutoAdvanceAfter';
+
+describe('getAutoAdvanceAfter', () => {
```
Comment (swissspidy) [inline packages/story-editor/src/output/utils/test/getAutoAdvanceAfter.js:23]: Tests for the animation duration part woud be great too

File: packages/story-editor/src/components/panels/shared/backgroundAudioPanelContent.js [hunk 5]
```diff
@@ -58,6 +59,7 @@ function BackgroundAudioPanelContent({
   backgroundAudio,
   updateBackgroundAudio,
   supportsCaptions = false,
+  supportsLooping = false,
```
Comment (swissspidy) [inline packages/story-editor/src/components/panels/shared/backgroundAudioPanelContent.js]: The more I look at it, the more I think the name of these two props is confusing.

WDYT about something `showLoopControl`/`showCpations` or `hasLoopControl`/`hasCaptions` instead?

File: packages/story-editor/src/output/test/page.js [hunk 6]
```diff
@@ -1539,6 +1578,35 @@ describe('Page output', () => {
 
       await expect(<PageOutput {...props} />).toBeValidAMPStoryPage();
     });
+
+    // TODO(#10338): Resolve question about poster requirement.
+    // eslint-disable-next-line jest/no-disabled-tests
+    it.skip('should produce valid output with none looping background audio', async () => {
```
Comment (swissspidy) [inline packages/story-editor/src/output/test/page.js]:
```suggestion
    it.skip('should produce valid output with non-looping background audio', async () => {
```

**Review by swissspidy — COMMENTED** [review]
Just a few nits otherwise this looks good

File: packages/story-editor/src/components/panels/design/pageBackgroundAudio/backgroundAudio.js [hunk 7]
```diff
@@ -82,7 +82,8 @@ function PageBackgroundAudioPanel() {
       <BackgroundAudioPanelContent
         backgroundAudio={backgroundAudio}
         updateBackgroundAudio={updateBackgroundAudio}
-        supportsCaptions={enhancedPageBackgroundAudio}
+        showCpations={enhancedPageBackgroundAudio}
```
Comment (swissspidy) [inline packages/story-editor/src/components/panels/design/pageBackgroundAudio/backgroundAudio.js]: Typo: `showCpations` -> `showCaptions`

**Review by swissspidy — APPROVED** [review]

**Pull request merged**